pytorch_lightning/utilities/imports.py

```diff
@@ -85,6 +85,25 @@ def _compare_version(package: str, op: Callable, version: str, use_base_version:
     return op(pkg_version, Version(version))
 
 
+class _RequirementAvaliable:
+    """Boolean-like class for check of requirement with extras and version specifiers."""
+
+    def __init__(self, requirement: str):
+        self.available = True
+        self.message = ""
+        try:
+            pkg_resources.require([requirement])
+        except Exception as ex:
+            self.available = False
+            self.message = f"Requirement '{requirement}' not met, {ex.__class__.__name__}: {ex}"
```

Comment:
just wondering if we should check this inside `__bool__` only once. then it will be checked only when it's required thus might reduce the import time #12786

Reply:
Good idea!

Reply:
>  where this class is instantiated the object is used right after

not all the time. It will still by imported even when user is not using LightningCli.
but anyways is fine, do it now or later for that issue.